microsoft/testsuites/performance/storageperf.py

```diff
@@ -187,6 +194,109 @@ def perf_storage_over_nfs_synthetic_tcp_4k(self, result: TestResult) -> None:
     def perf_storage_over_nfs_synthetic_udp_4k(self, result: TestResult) -> None:
         self._perf_nfs(result, protocol="udp")
 
+    @TestCaseMetadata(
+        description="""
+        This test case uses fio to test data disk performance with 4K block size.
+        """,
+        priority=3,
+        timeout=TIME_OUT,
+    )
+    def perf_storage_generic_fio_test(self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+        variables: Dict[str, Any],
+    ) -> None:
+        testcases = variables.get("fio_testcase_list", None)
+        if (not testcases):
+            testcases_file_path = os.path.join(
+                os.path.abspath(__file__).replace(os.path.basename(__file__), ""),
+                "default_fio_testcases.json",
+            )
+            with open(testcases_file_path, "r") as f:
+                testcases = json.loads(f.read())["testcases"]
+
+        log.debug(f"Given testcases with params : {testcases}")
+        fio = node.tools[Fio]
+        rm = node.tools[Rm]
+        fio_result = {}
+
+        for testcase in testcases:
+            test_name = testcase["testcase_name"]
+            fio_result[test_name] = {}
+            fio_result[test_name]["runtime_param"] = testcase
+
+            try:
+                path = node.get_working_path()
+
+                log.info(f"Executing the FIO testcase : {test_name}")
+
+                name = testcase.get("name", None)
```

Comment:
These can be handled in a separate class encapsulating these variables

Reply:
Sure we can do this. I will look for this.

Reply:
We have used perf_disk now to run testcase with default and runtime parameters. Please check